magento/magento2
# Confirmation emails have no FROM or FROM email address 2.2.4 #14952. …

Undone commit f2bfdd941c0c8c6c7c745156d1ce206e38c76820 which caused issue

### Description
Removed class  lib/internal/Magento/Framework/Mail/Template/TransportBuilderByStore and assosiated tests for it.
The class cause the issue #14952 and was introduced because in previous releases default class lib/internal/Magento/Framework/Mail/Template/TransportBuilder didn't have the parameter  $store  for function  public function setFromByStore($from, $store) Line 291.

### Fixed Issues (if relevant)

1. magento/magento2#14952: Confirmation emails have no FROM or FROM email address 2.2.4
2.  magento/magento2#24: Wrong sender name/email in invoice/shipment email in multistore


### Manual testing scenarios

1. When Magento has single store view: set values at the admin Store->Configuration->General->Store Email Addresses-> Sales Representative.  Make an order from **storefront** then check in the order success email "name from" and "email from" fields. It supposes to be matched to the admin configuration.

2. When Magento has single store view: set values at the admin Store->Configuration->General->Store Email Addresses-> Sales Representative.  Make an order from **admin panel** then check in the order success email "name from" and "email from" fields. It supposes to be matched to the admin configuration.

3. When Magento has multiple store views:  set different values at the admin  Store->Configuration->General->Store Email Addresses-> Sales Representative. 
4. Create a new order from the **storefront** for each store view and look at your email to confirm these values in order confirmation email. 
5. Create a new order from the **admin** for each store view and look at your email to confirm these values in order confirmation email. 


### Contribution checklist
